watershed_workflow/river_mesh.py

```diff
@@ -1181,3 +1181,72 @@ def adjust_hucs_for_river_corridor(hucs,
                                                                rc_points[rc_point_ind])
                     else:
                         rc_point_ind += 1
+
+
+def triangle_split_points(stream_triangles, river_corrs):
+    """Find additional points to split stream triangles.
+    
+    Parameters
+    ----------
+    stream_triangles : list of tuples
+        List of triangle vertices.
+    river_corrs : list of shapely.geometry.Polygon
+        List of river corridor polygons.
+    """
+    additional_points = []
+    for tri_verts in stream_triangles:
+        # Get midpoints of each edge
+        midpoints = [
+            ((tri_verts[0][0] + tri_verts[1][0])/2, (tri_verts[0][1] + tri_verts[1][1])/2),
+            ((tri_verts[1][0] + tri_verts[2][0])/2, (tri_verts[1][1] + tri_verts[2][1])/2),
+            ((tri_verts[2][0] + tri_verts[0][0])/2, (tri_verts[2][1] + tri_verts[0][1])/2)
+        ]
+        
+        # Check which midpoints lie on river corridors
+        on_corridor = []
+        off_corridor = []
+        for i, mp in enumerate(midpoints):
+            point = shapely.geometry.Point(mp)
+            if any(corridor.intersects(point.buffer(1)) for corridor in river_corrs):
+                on_corridor.append((i, mp))
+            else:
+                off_corridor.append((i, mp))
+        
+        # We expect exactly one midpoint to be off the corridor
+        if len(off_corridor) == 1:
+            # Find the vertex common to the two edges that lie on the corridor
+            off_edge_idx = off_corridor[0][0]
+            if off_edge_idx == 0:
+                center_point = tri_verts[2]
+                off_corridor_seg = shapely.geometry.LineString([tri_verts[0], tri_verts[1]])
+            elif off_edge_idx == 1:
+                center_point = tri_verts[0]
+                off_corridor_seg = shapely.geometry.LineString([tri_verts[1], tri_verts[2]])
+            else:  # off_edge_idx == 2
+                center_point = tri_verts[1]
+                off_corridor_seg = shapely.geometry.LineString([tri_verts[0], tri_verts[2]])
+            
+            # Find the additional point
+            additional_point = find_additional_point(off_corridor_seg, center_point, extension_factor=1)
+            additional_points.append(additional_point.coords[0])
+
+    return additional_points
+
+
+def find_additional_point(off_corridor_seg, center_point, extension_factor=1):
+    """Find an additional point on a common segment between a triangle and a river corridor.
+    
+    Parameters
+    ----------
+    off_corridor_seg : shapely.geometry.LineString
+        Triangle edge that is not on the river corridor.
+    center_point : tuple
+        Common vertex of on-corridor triangle edges.
+    extension_factor : float, optional
+        The factor by which to extend the vector from the center point to the midpoint of the off-corridor edge of the triangle.
+    """
+    mid_point = off_corridor_seg.interpolate(0.5, normalized=True)
```

Comment:
a second way of just calling shapely.utils.midpoint?

Reply:
ah yes haha, initially I was trying different locations, and then settled on midpoint.